BenMakesGames.PlayPlayMini/GameState.cs

```diff
@@ -1,78 +1,92 @@
+/// </summary>
+/// <remarks>
+/// If a game state needs access to a service, such as the <see cref="GameStateManager"/>, <see cref="GraphicsManager"/>, <see cref="SoundManager"/>, etc.,
 /// include the service in the game state's constructor arguments. The IoC container will automatically inject them.
+/// <para>
+/// <example>
+/// Example usage:
+/// <code>
+/// public sealed class MyGameState: GameState
+/// {
+///     private GameStateManager GSM { get; }
+///     private GraphicsManager Graphics { get; }
+///     private KeyboardManager Keyboard { get; }
 ///
-/// Example:
-///
-///     public sealed class MyGameState: GameState
+///     public MyGameState(GameStateManager gsm, GraphicsManager graphics, KeyboardManager keyboard)
 ///     {
-///         private GameStateManager GSM { get; }
-///         private GraphicsManager Graphics { get; }
-///         private KeyboardManager Keyboard { get; }
-///
-///         public MyGameState(GameStateManager gsm, GraphicsManager graphics, KeyboardManager keyboard)
-///         {
-///             GSM = gsm;
-///             Graphics = graphics;
-///             Keyboard = keyboard;
-///         }
-///
-///         public override void Update(GameTime gameTime)
-///         {
-///             if(Keyboard.IsKeyDown(Keys.Escape))
-///                 GSM.ChangeState&lt;MyPauseMenu&gt;(); // or maybe GSM.Exit();
-///         }
+///         GSM = gsm;
+///         Graphics = graphics;
+///         Keyboard = keyboard;
 ///     }
-/// </summary>
+///     
+///     public override void Input(GameTime gameTime)
+///     {
+///         if (Keyboard.IsKeyDown(Keys.Escape))
+///             GSM.ChangeState&lt;MyPauseMenu&gt;(); // or maybe GSM.Exit();
+///     }
+/// }
+/// </code>
+/// </example>
+/// </para>
+/// </remarks>
 public abstract class GameState
 {
     /// <summary>
-    /// This method is called once per frame. It's intended to be used to capture inputs from input devices,
-    /// such as the KeyboardManager, MouseManager, or a gamepad. Simpler games may get away with doing this
-    /// work in an Update method, instead.
+    /// Called when input should be captured from input devices. 
     /// </summary>
-    /// <param name="gameTime"></param>
+    /// <remarks>
+    /// Override this method to capture input from a service such as <see cref="KeyboardManager"/>, <see cref="MouseManager"/>, etc.
+    /// This method is called once per frame, immediately before <see cref="Update(GameTime)"/>.
+    /// </remarks>
+    /// <param name="gameTime">A <see cref="GameTime"/> instance containing the elapsed time since the last call to <see cref="Input"/> and the total time elapsed since the game started.</param>
     public virtual void Input(GameTime gameTime) { }
 
     /// <summary>
-    /// This method is called once per frame. It's intended to be used to run your game logic.
+    /// Called when the game state should update
```

Comment:
^ I worry about using the phrase "game state" here, since there are also `GameState`s (I recognize this is a problem of PPM's own-making, as I think "update your game state" is a reasonable phrase to say here).

another thought: I'd really like the documentation to speak to me, as a user, and make suggestions on what to do. I wonder if there are style guides specific to documentation; what I'm looking for feels adjacent to https://en.wikipedia.org/wiki/Human-centered_design, maybe? (and I don't think the old text necessarily did that very well, either.)

Reply:
Do you have a specific suggestion? I could say "Called when your game logic should run" but that is also vague.

For what it's worth though, if I were learning how to use a new framework, I'd be looking for  "getting started" documentation with examples rather than trying to figure out what to do PURELY from the doc blocks